src/main/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/validation/CardValidator.java

```diff
@@ -0,0 +1,80 @@
+package ch.uzh.ifi.hase.soprafs22.screwyourneighborserver.validation;
+
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+
+import ch.uzh.ifi.hase.soprafs22.screwyourneighborserver.entity.*;
+import ch.uzh.ifi.hase.soprafs22.screwyourneighborserver.repository.PlayerRepository;
+import java.util.Collection;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.rest.core.annotation.HandleBeforeSave;
+import org.springframework.data.rest.core.annotation.RepositoryEventHandler;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.parameters.P;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.HttpClientErrorException;
+
+@Component
+@RepositoryEventHandler
+public class CardValidator {
+
+  private final OldStateFetcher oldStateFetcher;
+  private final PlayerRepository playerRepository;
+
+  @Autowired
+  public CardValidator(PlayerRepository playerRepository, OldStateFetcher oldStateFetcher) {
+
+    this.playerRepository = playerRepository;
+    this.oldStateFetcher = oldStateFetcher;
+  }
+
+  @SuppressWarnings({"unused", "SpringElInspection", "ELValidationInspection"})
+  @HandleBeforeSave
+  @PreAuthorize("isOwnCard(#card)")
+  public void onUpdateCard(@P("card") Card card) {
+    Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+    Round newRound = card.getRound();
+    if (!(authentication.getPrincipal() instanceof Player)) {
```

Comment:
würde ich sagen braucht es nicht, da das schon mit "isOwnCard(#card)" gegeben ist

Reply:
Jaa, ich hatte das vor allem deswegen gemacht, damit ich nicht in irgendeinem Ausnahmefall unterhalb in eine Nullpointerexception reinlaufen könnte (bei der Zeile `Long playerId = ((Player) authentication.getPrincipal()).getId();`)